src/mixer/alsasoftvol.rs

```diff
@@ -0,0 +1,45 @@
+use super::Mixer;
+use super::AudioFilter;
+
+use alsa;
+
+#[derive(Clone)]
+pub struct AlsaSoftVol {
+    name: String
+}
+
+impl Mixer for AlsaSoftVol {
+    fn open(device: Option<String>) -> AlsaSoftVol {
+        let name = device.unwrap_or("default".to_string());
+        AlsaSoftVol {
+            name: name
+        }
+    }
+    fn start(&self) {
+    }
+    fn stop(&self) {
+    }
+    fn volume(&self) -> u16 {
+        let mixer = alsa::mixer::Mixer::new(&self.name, false).unwrap();
+        let selem_id = alsa::mixer::SelemId::new("Master", 0);
+        let selem = mixer.find_selem(&selem_id).unwrap();
+
+        let volume: i64 = selem.get_playback_volume(alsa::mixer::SelemChannelId::FrontLeft).unwrap();
+
+        volume as u16 * 256
```

Comment:
The factor 256 is only valid for ALSA mixers having a resolution of 256: 0xFFFF / resolution(256) = 256. If the resolution of the mixer is for example 128, the factor needs to be: 0xFFFF / resolution(128) = 512.